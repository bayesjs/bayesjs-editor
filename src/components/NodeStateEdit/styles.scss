.node-state-edit {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 3px 0px;

  &__button {
    height: 26px;
  }
}
