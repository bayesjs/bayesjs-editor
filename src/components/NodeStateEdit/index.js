import Button from '@components/Button';
import Icon from '@components/Icon';
import PropTypes from 'prop-types';
import React from 'react';
import { bem } from '@utils/styles';
import { getComponentTestId } from '@utils/test-utils';
import styles from './styles.scss';

const componentClassName = bem(styles);

const NodeStateEdit = ({ state, index, onDeleteState }) => {
  const onClick = () => onDeleteState({ state, index });

  return (
    <li className={componentClassName} data-testid={getComponentTestId('NodeStateEdit', state)}>
      <span title={state}>{state}</span>
      <Button
        className={componentClassName.element('button').toString()}
        onClick={onClick}
        title="Remover Estado"
        name="removeState"
      >
        <Icon name="trash" />
      </Button>
    </li>
  );
};

NodeStateEdit.propTypes = {
  state: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  onDeleteState: PropTypes.func.isRequired,
};

export default NodeStateEdit;
