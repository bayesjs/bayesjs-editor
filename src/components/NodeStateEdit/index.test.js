import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NodeStateEdit from './index';

const renderComponent = props =>
  render(
    <NodeStateEdit {...props} />,
  );

describe('NodeStateEdit Component', () => {
  const defaultProps = {
    state: 'State Test',
    index: 2,
    onDeleteState: jest.fn(),
  };
  let component;

  beforeEach(() => {
    defaultProps.onDeleteState.mockClear();
    component = renderComponent(defaultProps);
  });

  it('matches snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot();
  });

  describe('When clicked in remove button', () => {
    beforeEach(() => {
      const { getByTitle } = component;
      const removeButton = getByTitle('Remover Estado');

      fireEvent.click(removeButton);
    });

    it('calls onDeleteState function', () => {
      expect(defaultProps.onDeleteState).toBeCalledWith({
        state: defaultProps.state,
        index: defaultProps.index,
      });
    });
  });
});
