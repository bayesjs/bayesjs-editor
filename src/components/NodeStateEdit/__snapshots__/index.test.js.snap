// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeStateEdit Component matches snapshot 1`] = `
<li
  class=""
  data-testid="node-state-edit__state-test"
>
  <span
    title="State Test"
  >
    State Test
  </span>
  <button
    class="__button"
    data-testid="button__remove-state"
    title="Remover Estado"
    type="button"
  >
    <i
      class="fa fa-trash"
    />
  </button>
</li>
`;
